### ISSUE TITLE
UnknownZodTypeError mongo ObjectId

### ISSUE THREAD

**Issue by adborroto** [asteasolutions/zod-to-openapi#87]
When generating the document, I'm receiving an `UnknownZodTypeError`.

```
import { extendZodWithOpenApi } from '@asteasolutions/zod-to-openapi';
import { ObjectId } from 'mongoist';
import { z } from 'zod';

extendZodWithOpenApi(z);

const Schema = z.object({
  _id: z.string().or(z.instanceof(mongoist.ObjectId).openapi({example: "test"})).nullable(),
});
```
Not sure what I'm missing here. In the [documentation](https://github.com/asteasolutions/zod-to-openapi#unsupported-types) it says that I can provide a `type` via the `.openapi` but not sure what it means.
Thanks in advance.

**Comment by AGalabov** [comment]
@adborroto yes, `z.instanceof` is not supported. Not sure that rhere is anything we can do about it either. Will check. The way to provide an OpenApi type is to do `z.instanceof(mongoist.ObjectId).openapi({ type: <your-type-here> })`.

Basically the `.openapi` method allows you to provide any OpenApi specific information - type/examples/patterns etc.

I am not familiar with mongois.ObjectId but it sounds like it is a `type: 'string'`.

Please, let me know if that helps.

**Comment by samchungy** [comment]
tl;dr - we don't have a mapping to represent every possible type in Javascript -> openapi.

In this case you have a`mongoist.ObjectId` (class/object?) but there's no way for us to know what [datatype](https://swagger.io/docs/specification/data-models/data-types/) to assign to that field so you have to tell us what type you want in your doc.
